### PR TITLE
Creates tile_freeze/unfreeze/set_symbols routines for CUDA API 

### PR DESCRIPTION
**Pull request by bornaehsani**
This pull request addresses the following issues in the CUDA API: 
1. Issue #268 : Create hb_mc_device_tiles_freeze/unfreeze routines that sends packets to a list of tiles to freeze/unfreeze them
2. Issue #270 : Create hb_mc_device_tiles_set_symbols routine that replaces the large section inside hb_mc_program_load and hb_mc_tile_group_initialize_tiles, sending packets to all tiles in a list to set their binary symbols. 
3. Issue #269 : Removed unnecessary uses of malloc and replaced with stack allocation. Also changed uses of "new" to "malloc"
4. Issue #267 : Removed unnecessary use of uint8_t in hb_mc_wait_for_tile_group_finish_any, other uses of uint8_t in deprecated functions will be addressed in a separate PR.
5. Issue #272 : Moved all hb_mc_tile_set_***_symbols from bsg_manycore_tile.cpp/h to bsg_manycore_cuda.cpp/h
6. Issue #271 : Replaced hb_mc_memcpy_to_device/host to upper case HB_MC_MEMCPY_TO_DEVICE/HOST
7. Issue #279 fixed.
8. Minor bug fix in bsg_manycore_origin_eva_map.cpp 
9. For all hb_mc_tile_set_origin/coord/id/tg_id/grid_dim routines, used hb_mc_manycore_eva_write instead of eva_to_npa and hb_mc_manycore_write
 

Rebased and regression tests pass. Ready to merge.

### CODE REVIEW

File: cl_manycore/libraries/bsg_manycore_cuda.cpp [hunk 1]
```diff
@@ -1365,3 +1258,599 @@ static hb_mc_idx_t hb_mc_get_tile_id (hb_mc_coordinate_t origin, hb_mc_dimension
...
+					const hb_mc_coordinate_t *coord,
+					const hb_mc_coordinate_t *origin){
+
+	int error;
+	const hb_mc_config_t *cfg = hb_mc_manycore_get_config (mc); 
+
+	hb_mc_eva_t org_x_eva, org_y_eva;
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_grp_org_x", &org_x_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_grp_org_x eva.\n", __func__);
```
Comment (max-ruttenberg) [inline cl_manycore/libraries/bsg_manycore_cuda.cpp]: that's spelled "acquire"

Comment (max-ruttenberg) [reply cl_manycore/libraries/bsg_manycore_cuda.cpp]: C needs a spell checker for strings; I mean really it's 2019!

File: cl_manycore/libraries/bsg_manycore_cuda.cpp [hunk 2]
```diff
@@ -1365,3 +1258,599 @@ static hb_mc_idx_t hb_mc_get_tile_id (hb_mc_coordinate_t origin, hb_mc_dimension
...
+	error = hb_mc_loader_symbol_to_eva(bin, bin_size, "__bsg_y", &bsg_y_eva); 
+	if (error != HB_MC_SUCCESS) { 
+		bsg_pr_err("%s: failed to aquire __bsg_y eva.\n", __func__);
+		return HB_MC_NOTFOUND;
+	}
+
+
+	hb_mc_npa_t bsg_x_npa, bsg_y_npa;
+	size_t sz;
+	error = hb_mc_eva_to_npa(cfg, map, coord, &bsg_x_eva, &bsg_x_npa, &sz);
```
Comment (max-ruttenberg) [inline cl_manycore/libraries/bsg_manycore_cuda.cpp]: There's a function declared in `bsg_manycore_eva.h` called `bsg_manycore_eva_write()` that does Eva translation and writing all at once. You don't need to change your code but that's something to know for future reference.

**Comment by max-ruttenberg** [comment]
Does this compile?

File: cl_manycore/libraries/bsg_manycore_cuda.cpp (hunk 2 above)
Comment (drichmond) [reply cl_manycore/libraries/bsg_manycore_cuda.cpp]: I would argue, that where we can, we should use the appropriate functions.

Comment (max-ruttenberg) [reply cl_manycore/libraries/bsg_manycore_cuda.cpp]: OK. @bornaehsani please address this.

File: cl_manycore/libraries/bsg_manycore_cuda.cpp (hunk 1 above)
Comment (bornaehsani) [reply cl_manycore/libraries/bsg_manycore_cuda.cpp]: Ha yeah agreed! Thanks for pointing it out, I'll take care of it!

File: cl_manycore/libraries/bsg_manycore_cuda.cpp (hunk 2 above)
Comment (bornaehsani) [reply cl_manycore/libraries/bsg_manycore_cuda.cpp]: Oh you're right! Thanks for pointing it out, I'll take care of them!

**Comment by bornaehsani** [comment]
> 
> 
> Does this compile?

Yeah, it doesn't compile for you?

**Comment by drichmond** [comment]
Does this compile and pass on F1?

**Comment by bornaehsani** [comment]
@drichmond  @max-ruttenberg , compiles and F1 passes. I believe all issues are addressed now. Let me know what you think, thanks a lot!

**Comment by drichmond** [comment]
Hi Borna, what commits on BSG Manycore and BaseJump is this passing with?

**Comment by bornaehsani** [comment]
Hi Dustin, the F1 passes with AGFI image v1.0.0, bsg_manycore commit 8241449, basejump stl commit 2bb5555. 
I just tested and it also passes with AGFI image v1.0.1, bsg_manycore commit 8241449, basejump_stl commit 2bb5555

**Comment by drichmond** [comment]
Hmm. Now there's a conflict with Makefile.common

**Comment by bornaehsani** [comment]
Yeah I fixed it, just running a quick cosim to make sure everything is in check

**Comment by bornaehsani** [comment]
OK we are good to go!

**Comment by drichmond** [comment]
Thanks Borna!

Good commit messages in this one

**Comment by bornaehsani** [comment]
Thanks a lot, I'm starting to learn :D

**Pull request merged**